### PR TITLE
Extending string tests

### PR DESCRIPTION
**Pull request by davidpiuva**
Made wrapper functions for performing tests and using them to run lots of test cases.

### CODE REVIEW

**Comment by davidpiuva** [comment]
@sebastianbaginski Peer review?

**Comment by sebastianbaginski** [comment]
This looks great.

File: test/system/StringTest.ooc [hunk 1]
```diff
@@ -149,6 +192,114 @@ StringTest: class extends Fixture {
...
+			reverseTest("", "")
+			reverseTest(" ", " ")
+			reverseTest("1", "1")
+			reverseTest("12", "21")
+			reverseTest("a", "a")
+			reverseTest("ab", "ba")
+			reverseTest("abc", "cba")
+			reverseTest("Test", "tseT")
+			reverseTest("Test!", "!tseT")
+			reverseTest("marcus", "sucram")
```
Comment (marcusnaslund) [inline test/system/StringTest.ooc]: 5 bonus points for keeping my name in the test.

**Comment by marcusnaslund** [comment]
Make the helper functions static and add `This` to the calls, and I'm happy too.

**Comment by davidpiuva** [comment]
Updated

File: test/system/StringTest.ooc [hunk 2]
```diff
@@ -149,6 +192,114 @@ StringTest: class extends Fixture {
...
+			This reverseTest("Testing", "gnitseT")
+			This reverseTest("Reversing a very long sentence just in case of weird upper limits when allocating the result.", ".tluser eht gnitacolla nehw stimil reppu driew fo esac ni tsuj ecnetnes gnol yrev a gnisreveR")
+		})
+		this add("findAll", func {
+			This findTest("", "")
+			This findTest("a", "")
+			This findTest("", "a")
+			This findTest("abab", "ab", 0, 2)
+			This findTest("abcab", "ab", 0, 3)
+			This findTest("ababa", "aba", 0) // Finding following overlaps would make it useless for split
```
Comment (marcusnaslund) [inline test/system/StringTest.ooc]: Does `String` not have any method which *would* find two indices?

Comment (davidpiuva) [inline test/system/StringTest.ooc]: I have not found such a function.

Comment (davidpiuva) [inline test/system/StringTest.ooc]: Regex pattern matching works this way too because otherwise, the search results would be overlapping.

Comment (marcusnaslund) [inline test/system/StringTest.ooc]: Sure, I'm not saying there's anything wrong with it, especially for `split` or `replace`, but there are instances where overlaps are *also* acceptable.

Comment (marcusnaslund) [inline test/system/StringTest.ooc]: Nothing for this PR though.

**Pull request merged**